iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -628,6 +628,64 @@ static LogicalResult setRootConfig(
       DispatchLoweringPassPipeline::CPUDoubleTilingExpert);
 }
 
+/// Sets the lowering configuration for linalg.conv_2d_nhwc_hwcf operations.
+static LogicalResult setRootConfig(
+    FuncOp entryPointFn, linalg::Conv2DNhwcHwcfOp convOp,
+    ArrayRef<LoopTilingAndDistributionInfo> tiledLoops) {
+  auto linalgOp = cast<linalg::LinalgOp>(convOp.getOperation());
+  // Use the default distribution for the matmul loops.
+  unsigned numLoops = linalgOp.getNumLoops();
+  int64_t vectorSize =
+      getVectorSize(entryPointFn, convOp.getResult(0).getType());
+  SmallVector<int64_t> minTileSizes(numLoops, 1);
+  SmallVector<int64_t> maxTileSizes(numLoops, defaultWorkgroupTileSize);
+
+  // Set the flow level tiling to the default.
+  OpBuilder builder(convOp.getContext());
+  builder.setInsertionPoint(convOp);
+  SmallVector<Range> iterationDomain =
+      linalgOp.createLoopRanges(builder, convOp.getLoc());
+  auto partitionableLoopsInterfaceOp =
+      cast<IREE::Flow::PartitionableLoopsInterface>(convOp.getOperation());
+  SmallVector<int64_t> flowTileSizes = getDefaultDistributedLevelTileSizes(
+      iterationDomain, partitionableLoopsInterfaceOp, minTileSizes,
+      maxTileSizes);
+
+  // Shapes of N, OH, OW, OC, KH, KW, IC
+  Optional<SmallVector<int64_t, 4>> shapes = linalgOp.getStaticLoopRanges();
+  assert(shapes.hasValue() &&
+         "something went wrong when inferring loop ranges");
+
+  SmallVector<int64_t> l1TileSizes = {1, 1, 8, vectorSize * 2, 1, 1, 8};
+  for (auto i : llvm::seq<unsigned>(0, l1TileSizes.size())) {
+    if (flowTileSizes[i]) {
+      l1TileSizes[i] = getMaxTileSize(0, flowTileSizes[i], l1TileSizes[i],
```

Comment:
Why `vectorSizeVal` is 1 here?

Reply:
It is a fallback value that make sure they are vectorized. I added another variable to the function. I'll revisit if we want this for all the kernels (99% certainly sure we want :D). Since this PR is mainly for convolution vectorization, I'd like to address these cleanups in followup PRs.